aspnetcore/fundamentals/middleware.md

```diff
@@ -188,17 +188,17 @@ app.Map("/level1/level2", HandleMultiSeg);
 
 ## Built-in middleware
 
-ASP.NET Core ships with the following middleware components:
-
-| Middleware | Description |
-| ----- | ------- |
-| [Authentication](xref:security/authentication/identity) | Provides authentication support. |
-| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. |
-| [Response Caching](xref:performance/caching/middleware) | Provides support for caching responses. |
-| [Response Compression](xref:performance/response-compression) | Provides support for compressing responses. |
-| [Routing](xref:fundamentals/routing) | Defines and constrains request routes. |
-| [Session](xref:fundamentals/app-state) | Provides support for managing user sessions. |
-| [Static Files](xref:fundamentals/static-files) | Provides support for serving static files and directory browsing. |
+ASP.NET Core ships with the following middleware components, as well as a description of the order in which they run:
+
+| Middleware | Description | Ordering |
+| ----- | ------- | ------- |
+| [Authentication](xref:security/authentication/identity) | Provides authentication support. | Before `HttpContext.User` is needed. Terminal for OAuth callbacks. |
+| [CORS](xref:security/cors) | Configures Cross-Origin Resource Sharing. | Before components that use CORS |
```

Comment:
Diagnostics is missing from the list. So is HttpOverrides/ForwardedHeaders, RequestLocalization, and WebSockets.